lte/cloud/go/services/subscriberdb/storage/digest.go

```diff
@@ -0,0 +1,222 @@
+/*
+ Copyright 2020 The Magma Authors.
+
+ This source code is licensed under the BSD-style license found in the
+ LICENSE file in the root directory of this source tree.
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package storage
+
+import (
+	"database/sql"
+
+	"magma/orc8r/cloud/go/clock"
+	"magma/orc8r/cloud/go/sqorc"
+
+	"github.com/Masterminds/squirrel"
+	"github.com/pkg/errors"
+)
+
+type digestLookup struct {
+	db      *sql.DB
+	builder sqorc.StatementBuilder
+}
+
+type digestInfo struct {
+	network         string
+	digest          string
+	lastUpdatedTime int64
+}
+
+type DigestInfoSet []digestInfo
+
+// GetNetworks returns a list of network IDs for all digests in DigestInfoSet.
+func (digestInfoSet DigestInfoSet) GetNetworks() []string {
+	ret := []string{}
+	for _, digestInfo := range digestInfoSet {
+		ret = append(ret, digestInfo.network)
+	}
+	return ret
+}
+
+type DigestLookup interface {
+	// Initialize the backing store.
+	Initialize() error
+
+	// GetDigests returns a list of digests of that satisfy the filtering criteria
+	// specified by the arguments.
+	// Caveats:
+	// 1. If networks is empty, returns digests for all networks.
+	// 2. earliestUpdateTime is recorded in unix seconds. If a positive integer,
+	// filter for all digests that were last updated earlier than this time
+	// (i.e. outdated digests); otherwise, the time filter is not applied.
```

Comment:
From previous comment: no need to add the "if a positive integer...", because callers can just use `0` to "effectively" read digests regardless of time
```suggestion
	// 2. earliestUpdateTime is recorded in unix seconds. Filters for all digests that
	// were last updated earlier than this time. Use zero to ignore the filter.
```